### PR TITLE
[Event Hubs] Diagnostics and validation tweaks

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to fix a couple of formatting issues in the ETW events and to enhance the information provided when links are opened and when the processor begins processing a partition.  In addition, startup validation logic has been adjusted to use a custom set of options when testing read permissions to avoid the default prefetch count being used and transfer of more data than is needed from the service.

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Azure.Messaging.EventHubs.Processor.csproj [hunk 1]
```diff
@@ -14,7 +14,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Azure.Messaging.EventHubs" />
+    <PackageReference Include="Azure.Messaging.EventHubs"/>
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Azure.Messaging.EventHubs.Processor.csproj:17]:
```suggestion
    <PackageReference Include="Azure.Messaging.EventHubs" />
```

this was unintentional and will be reverted after the test runs complete.

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Processor/EventProcessorClientLiveTests.cs [hunk 2]
```diff
@@ -4,10 +4,12 @@
 using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.IO;
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Processor/EventProcessorClientLiveTests.cs:7]: This was left dangling from testing and will be reverted after test runs complete.

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Diagnostics/EventHubsEventSource.cs [hunk 3]
```diff
@@ -2157,23 +2171,27 @@ public void UnexpectedException(string errorMessage)
...
+                                                                                     string authorIdentifier,
+                                                                                     DateTimeOffset? lastModified)
         {
             if (IsEnabled())
             {
+                authorIdentifier ??= "None";
+
                 var selectionBasedOn = checkpointUsed ? $"checkpoint" : "default";
-                WriteEvent(105, partitionId ?? string.Empty, identifier ?? string.Empty, eventHubName ?? string.Empty, consumerGroup ?? string.Empty, startingPosition ?? string.Empty, selectionBasedOn, authorIdentifier ?? string.Empty, lastModified ?? default);
+                var lastModifiedString = lastModified.HasValue ? lastModified.Value.ToString("yyyy-mm-ddTHH:mm:ss.fffZ", CultureInfo.InvariantCulture) : "None";
```
Comment (m-redding) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Diagnostics/EventHubsEventSource.cs:2192]: Thank you for fixing this!

**Review by m-redding — APPROVED** [review]

**Comment by jsquire** [comment]
CI build is failing due to a known issue with libraries that split pipelines to separate packages as it is not taking into account which packages are built for this specific SDK type.    Overriding and merging.

**Pull request merged**